Nanoleaf-wpf/Views/Options/OptionsWindow.xaml.cs

```diff
@@ -259,5 +275,57 @@ private string FullNameForCulture(string text)
 
             return null;
         }
+
+        private void AddColor_Click(object sender, RoutedEventArgs e)
+        {
+            var color = ColorPicker.SelectedColor;
+            var name = EffectTextBox.Text;
+
+            if (UserSettings.Settings.CustomEffects != null && UserSettings.Settings.CustomEffects.Any(effect => effect.EffectName == name))
+            {
+                PopupCreator.Error(Options.Resources.NameTaken);
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                PopupCreator.Error(Options.Resources.NoNameProvided);
+                return;
+            }
+
+            if (color == null)
+            {
+                PopupCreator.Error(Options.Resources.NoColorSelected);
+                return;
+            }
+
+            var customEffect = new UserCustomColorEffect()
+            {
+                Color = Color.FromArgb(color.Value.A, color.Value.R, color.Value.G, color.Value.B),
+                EffectName = name
+            };
+
+            if (UserSettings.Settings.CustomEffects == null)
+            {
+                UserSettings.Settings.CustomEffects = new List<UserCustomColorEffect>();
+            }
+
+            // Add color in settings.
+            UserSettings.Settings.CustomEffects.Add(customEffect);
```

Comment:
This introduces inconsistent behaviour. If you cancel the options window, any color added or deleted is still saved. I think it is better to set the lsit of custom effects on save, such that it is in line with the behaviour of the other settings